packages/common/src/versionedTypeHelpers.ts

```diff
@@ -0,0 +1,104 @@
+import {
+    AccountInfo,
+    AccountInfoBakerV0,
+    AccountInfoBakerV1,
+    Authorizations,
+    AuthorizationsV1,
+    BlockInfo,
+    BlockInfoV0,
+    BlockInfoV1,
+    ChainParameters,
+    ChainParametersV0,
+    ChainParametersV1,
+    ChainParametersV2,
+    ConsensusStatus,
+    ConsensusStatusV0,
+    ConsensusStatusV1,
+    ElectionInfo,
+    ElectionInfoV0,
+    ElectionInfoV1,
+    Keys,
+    KeysV0,
+    KeysV1,
+    StakePendingChange,
+    StakePendingChangeV0,
+    StakePendingChangeV1,
+    UpdateQueues,
+    UpdateQueuesV0,
+    UpdateQueuesV1,
+} from './types';
+
+export const isBakerAccountV0 = (ai: AccountInfo): ai is AccountInfoBakerV0 =>
+    (ai as AccountInfoBakerV1).accountBaker?.bakerPoolInfo === undefined;
```

Comment:
It seems wrong to cast in all of these predicates, since the point of the function is to determine the type of the argument. It will probably be fine, but you do lose some type safety here and the reader now have to think about it in order to believe it works. I generally prefer not having to think :smiling_face_with_tear:

Reply:
I agree.. It would be better with a clear distinction using a `type` member on all of these (which also more precisely translates the GRPC type). This is more of a leftover from the grpcv1 API. I don't know if I want to include this in the scope of this feature however.

Ideally, we'd remove the deprecated `JSONRPCClient` in which case we'd only have to deal with the types from the GRPCv2 API. I think the body of work included in what you're describing would fit nicely with that?

Reply:
Yes, let's say that

Reply:
@limemloh for reference, I've created this issue: https://github.com/Concordium/concordium-node-sdk-js/issues/212